plugin/src/main/java/org/opensearch/ml/model/MLModelAndNodeManager.java

```diff
@@ -0,0 +1,294 @@
+
+        // if we don't need to reload automatically, just return without doing anything
+        if (!enableAutoReLoadModel) {
+            return;
+        }
+
+        // At opensearch startup, get local node id, if not ml node,we ignored, just return without doing anything
+        String localNodeId = clusterService.localNode().getId();
+        DiscoveryNode node = nodeHelper.getNode(localNodeId);
+        if (!MLNodeUtils.isMLNode(node)) {
+            return;
+        }
+
+        // According to the node id to query the number of retries, if more than 2 (the maximum number of retries), we do not need to retry,
+        // that the number of unsuccessful reload has reached the maximum number of times, do not need to reload
+        Integer reTryTimes = 0;
+        if (isExistedIndex(ML_MODEL_RELOAD_INDEX)) {
+            reTryTimes = getReTryTimes(localNodeId);
+        }
+        if (reTryTimes > ML_MODEL_RELOAD_MAX_RETRY_TIMES) {
+            throw new RuntimeException("have exceeded max retry times, always failure");
+        }
+
+        // auto reload all models of this local node, if it fails, reTryTimes+1, if it succeeds, reTryTimes is cleared to 0
+        autoReLoadModelByNodeId(localNodeId);
+    }
+
+    @VisibleForTesting
+    void autoReLoadModelByNodeId(String nodeId) {
+        SearchRequest searchRequest = new SearchRequest(ML_TASK_INDEX);
+        SearchResponse response = client.execute(SearchAction.INSTANCE, searchRequest).actionGet();
+
+        if (response == null) {
+            return;
+        }
+        SearchHits searchHits = response.getHits();
+        if (searchHits == null) {
+            return;
+        }
+        SearchHit[] hits = searchHits.getHits();
+        if (CollectionUtils.isEmpty(hits)) {
+            return;
+        }
+
+        Integer reTryTimes = 0;
+        for (SearchHit hit : hits) {
+            try (XContentParser parser = createXContentParserFromRegistry(xContentRegistry, hit.getSourceRef())) {
+                ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                MLTask mlTask = MLTask.parse(parser);
+
+                String workerNode = mlTask.getWorkerNode();
+                MLTaskType mlTaskType = mlTask.getTaskType();
+                MLTaskState mlTaskState = mlTask.getState();
+
+                if (workerNode.contains(nodeId) && mlTaskType == MLTaskType.LOAD_MODEL && mlTaskState == MLTaskState.COMPLETED) {
+                    autoReLoadModelByNodeAndModelId(nodeId, mlTask.getModelId());
+                }
+            } catch (Exception e) {
+                reTryTimes++;
+                // Store the latest value of the reTryTimes and node id under the index ".plugins-ml-model-reload"
+                saveLatestReTryTimes(nodeId, reTryTimes);
+                log.error("ML Task id {} failed to parse, the reason is: {}", hit.getId(), e);
+                throw new RuntimeException(e);
```

Comment:
No need to throw exception and `saveLatestRetryTimes` can be delete from here as well.

Reply:
I think we can remove exception code,but the saveLatestRetryTimes method should be here.
otherwise,we don't know how many times we have failed to auto reload model.